client/packages/common/src/hooks/useConfirmOnLeaving/useConfirmOnLeaving.ts

```diff
@@ -1,23 +1,20 @@
 import { useContext, useEffect, useRef } from 'react';
 import { UNSAFE_NavigationContext as NavigationContext } from 'react-router-dom';
-import type { History } from 'history';
 import { useTranslation } from '@common/intl';
 
+const promptUser = (e: BeforeUnloadEvent) => {
```

Comment:
the method was being recreated on rerender and so the removeEventListener wasn't removing it